Mathlib/AlgebraicGeometry/EllipticCurve/Projective.lean

```diff
@@ -1673,11 +1694,250 @@ end Point
+    (W'.map f).toProjective.addX (f ∘ P) (f ∘ Q) = f (W'.addX P Q) := by
+  simp only [addX]
+  map_simp
+
+@[simp]
+lemma map_negAddY (P Q : Fin 3 → R) :
+    (W'.map f).toProjective.negAddY (f ∘ P) (f ∘ Q) = f (W'.negAddY P Q) := by
+  simp only [negAddY]
+  map_simp
+
+@[simp]
+lemma map_addY (P Q : Fin 3 → R) :
+    (W'.map f).toProjective.addY (f ∘ P) (f ∘ Q) = f (W'.addY P Q) := by
+  simp only [addY, negY_eq, map_negAddY, map_addX, map_addZ]
+  map_simp
+
+@[simp]
+lemma map_addXYZ (P Q : Fin 3 → R) :
+    (W'.map f).toProjective.addXYZ (f ∘ P) (f ∘ Q) = f ∘ addXYZ W' P Q := by
+  simp only [addXYZ, map_addX, map_addY, map_addZ, comp_fin3]
+
+variable {f} in
+@[simp]
+protected lemma map_add {K : Type v} [Field K] (f : F →+* K) {P Q : Fin 3 → F}
+    (hP : W.Nonsingular P) (hQ : W.Nonsingular Q) :
+    (W.map f).toProjective.add (f ∘ P) (f ∘ Q) = f ∘ W.add P Q := by
+  by_cases h : P ≈ Q
+  · rw [add_of_equiv <| (comp_equiv_comp f hP hQ).mpr h, add_of_equiv h, map_dblXYZ]
+  · rw [add_of_not_equiv <| h.comp (comp_equiv_comp f hP hQ).mp, add_of_not_equiv h, map_addXYZ]
+
+end Map
+
+section BaseChange
+
+/-! ### Base changes across algebra homomorphisms -/
+
+variable {R : Type r} [CommRing R] (W' : Projective R) {S : Type s} [CommRing S] [Algebra R S]
+  {A : Type u} [CommRing A] [Algebra R A] [Algebra S A] [IsScalarTower R S A]
+  {B : Type v} [CommRing B] [Algebra R B] [Algebra S B] [IsScalarTower R S B] (f : A →ₐ[S] B)
+
+lemma baseChange_polynomial : (W'.baseChange B).toProjective.polynomial =
+    MvPolynomial.map f (W'.baseChange A).toProjective.polynomial := by
+  rw [← map_polynomial, map_baseChange]
+
+variable {g} in
+lemma baseChange_equation (hf : Function.Injective f) (P : Fin 3 → A) :
+    (W'.baseChange B).toProjective.Equation (f ∘ P) ↔
+      (W'.baseChange A).toProjective.Equation P := by
+  rw [← map_equation hf, AlgHom.toRingHom_eq_coe, map_baseChange, RingHom.coe_coe]
+
+lemma baseChange_polynomialX : (W'.baseChange B).toProjective.polynomialX =
+    MvPolynomial.map f (W'.baseChange A).toProjective.polynomialX := by
+  rw [← map_polynomialX, map_baseChange]
+
+lemma baseChange_polynomialY : (W'.baseChange B).toProjective.polynomialY =
+    MvPolynomial.map f (W'.baseChange A).toProjective.polynomialY := by
+  rw [← map_polynomialY, map_baseChange]
+
+lemma baseChange_polynomialZ : (W'.baseChange B).toProjective.polynomialZ =
+    MvPolynomial.map f (W'.baseChange A).toProjective.polynomialZ := by
+  rw [← map_polynomialZ, map_baseChange]
+
+variable {f} in
+lemma baseChange_nonsingular (hf : Function.Injective f) (P : Fin 3 → A) :
+    (W'.baseChange B).toProjective.Nonsingular (f ∘ P) ↔
+      (W'.baseChange A).toProjective.Nonsingular P := by
+  rw [← map_nonsingular hf, AlgHom.toRingHom_eq_coe, map_baseChange, RingHom.coe_coe]
+
+lemma baseChange_negY (P : Fin 3 → A) :
+    (W'.baseChange B).toProjective.negY (f ∘ P) = f ((W'.baseChange A).toProjective.negY P) := by
+  erw [← map_negY, map_baseChange]
```

Comment:
All these `erw`s are probably pointing at a missing lemma. Can you see if that is the case?

Reply:
Apparently not! I just have to add `RingHom.coe_coe` in each `rw`.